src/main/kotlin/no/nav/faktureringskomponenten/service/FakturaGenerator.kt

```diff
@@ -42,28 +43,41 @@ open class FakturaGenerator(private val fakturalinjeGenerator: FakturaLinjeGener
         return samletFakturaListe
     }
 
-    private fun faktureringSluttDatoFra(dato: LocalDate, intervall: FakturaserieIntervall): LocalDate {
-        if (intervall == FakturaserieIntervall.MANEDLIG) return dato.withDayOfMonth(dato.lengthOfMonth())
-        return dato.withMonth(dato[IsoFields.QUARTER_OF_YEAR] * 3).with(TemporalAdjusters.lastDayOfMonth())
+    private fun faktureringSluttDatoFra(startDato: LocalDate, intervall: FakturaserieIntervall): LocalDate {
+        var sluttDato = if (intervall == FakturaserieIntervall.MANEDLIG) {
+            startDato.withDayOfMonth(startDato.lengthOfMonth())
+        } else {
+            startDato.withMonth(startDato[IsoFields.QUARTER_OF_YEAR] * 3).with(TemporalAdjusters.lastDayOfMonth())
+        }
+
+        if (startDato.year != sluttDato.year) {
+            sluttDato = LocalDate.of(startDato.year, 12, 31)
+        }
+
+        return sluttDato
     }
 
     private fun lagFakturaLinjerForPeriode(
         gjeldendeFaktureringStartDato: LocalDate,
         gjeldendeFaktureringSluttDato: LocalDate,
         fakturaseriePerioder: List<FakturaseriePeriode>,
         sluttDatoForHelePerioden: LocalDate
-    ): List<FakturaLinje> {
-        val fakturaLinjerForPeriode = fakturalinjeGenerator.lagFakturaLinjer(
-            perioder = fakturaseriePerioder,
-            faktureringFra = gjeldendeFaktureringStartDato,
-            faktureringTil = sluttDatoFra(gjeldendeFaktureringSluttDato, sluttDatoForHelePerioden)
-        )
-        return fakturaLinjerForPeriode
-    }
+    ): List<FakturaLinje> = fakturalinjeGenerator.lagFakturaLinjer(
+        perioder = fakturaseriePerioder,
+        faktureringFra = gjeldendeFaktureringStartDato,
+        faktureringTil = sluttDatoFra(gjeldendeFaktureringSluttDato, sluttDatoForHelePerioden)
+    )
 
     private fun sluttDatoFra(sisteDagAvPeriode: LocalDate, sluttDatoForHelePerioden: LocalDate) =
         if (sisteDagAvPeriode > sluttDatoForHelePerioden) sluttDatoForHelePerioden else sisteDagAvPeriode
 
+    private fun skalLageFakturaForPeriode(
+        dagensDato: LocalDate,
+        gjeldendeFaktureringSluttDato: LocalDate
+    ): Boolean = dagensDato <= gjeldendeFaktureringSluttDato || erSisteDagIÅret(gjeldendeFaktureringSluttDato)
+
+    private fun erSisteDagIÅret(dato: LocalDate): Boolean = dato.month == Month.DECEMBER && dato.dayOfMonth == 31
```

Comment:
Ikke alltid 31 dager, bruk f.eks LocalDate sin innebygde funksjon for å finne ut siste dag i året.

```
fun getLastDayOfYear(year: Int): LocalDate {
    return LocalDate.ofYearDay(year, LocalDate.of(year, 1, 1).lengthYear())
}
```

Reply:
Tror det er safe å anta at det er 31 dager i desember 😄

Reply:
Jeg vet ikke lengre..

![image](https://github.com/navikt/faktureringskomponenten/assets/10921090/bf7ac2a1-0369-483c-8782-8d0b7c212f7f)

Reply:
Det har vært 31 dager i desember ganske lenge, men jeg har ikke noe problem med å lage en annen funksjon

Reply:
Behold det, synes det ikke er like lett å lese mitt forslag som ditt

Reply:
![image](https://github.com/navikt/faktureringskomponenten/assets/45845087/7128d86d-940b-4e62-bdfe-2aa37916cc6d)

Reply:
AI er ikke så smart noen ganger